server.go

```diff
@@ -294,6 +309,17 @@ func makeUDPBuffer(size int) func() interface{} {
 func (srv *Server) init() {
 	srv.queue = make(chan *response)
 
+	const workersCount = 100
+	for i := 0; i < workersCount; i++ {
+		go func() {
+			for msg := range srv.queue {
+				msg.wg.Add(1)
```

Comment:
You can’t call `Add` from inside a goroutine because will end up concurrent with `Wait`.